app/src/main/kotlin/at/bitfire/davdroid/ui/AppSettingsModel.kt

```diff
@@ -107,4 +113,58 @@ class AppSettingsModel @Inject constructor(
 
     val pushEndpoint = preference.unifiedPushEndpointFlow()
 
+    private val _pushDistributor = MutableStateFlow<String?>(null)
+    val pushDistributor get() = _pushDistributor.asStateFlow()
+
+    private val _pushDistributors = MutableStateFlow<List<PushDistributorInfo>?>(null)
+    val pushDistributors get() = _pushDistributors.asStateFlow()
+
+    fun updatePushDistributor(pushDistributor: String) {
+        viewModelScope.launch(Dispatchers.IO) {
+            UnifiedPush.saveDistributor(context, pushDistributor)
+            UnifiedPush.registerApp(context)
+            _pushDistributor.emit(pushDistributor)
+        }
+    }
+
+
+    init {
+        viewModelScope.launch(Dispatchers.IO) {
+            val savedPushDistributor = UnifiedPush.getSavedDistributor(context)
+            _pushDistributor.emit(savedPushDistributor)
+
+            val pushDistributors = UnifiedPush.getDistributors(context)
+
+            val pushDistributorsData = mutableMapOf<String, Pair<String, Drawable>>()
+            for (pushDistributor in pushDistributors) {
+                try {
+                    val applicationInfo = pm.getApplicationInfo(pushDistributor, 0)
+                    val label = pm.getApplicationLabel(applicationInfo).toString()
+                    val icon = pm.getApplicationIcon(applicationInfo)
+                    pushDistributorsData[pushDistributor] = label to icon
+                } catch (_: PackageManager.NameNotFoundException) {
+                    // the app is not available for some reason, do not add anything
+                }
+            }
+            _pushDistributors.emit(
+                pushDistributors.map { distributor ->
+                    val data = pushDistributorsData[distributor]
+                    PushDistributorInfo(distributor, data?.first, data?.second)
+                }
+            )
+
+            // If there's already a distributor configured, register the app
+            UnifiedPush.getAckDistributor(context)?.let {
+                UnifiedPush.registerApp(context)
+            }
```

Comment:
What's the purpose of this?

Is it when there's only one distributor, is it then automatically configured?

Reply:
It was in the docs of UnifiedPush. The website seems to be down right now, so I can't link it here.

I've added now an automatic configuration